src/hotspot/share/code/codeCache.cpp

```diff
@@ -1194,7 +1194,7 @@ void CodeCache::initialize() {
     initialize_heaps();
   } else {
     // Use a single code heap
-    FLAG_SET_ERGO(NonNMethodCodeHeapSize, 0);
+    FLAG_SET_ERGO(NonNMethodCodeHeapSize, (uintx)os::vm_page_size());
```

Comment:
Sanity check: this code does execute after `os::init()` doesn't it?

Reply:
Yes. In fact, `FLAG_SET_ERGO` internally calls the constraint function for this flag, `VMPageSizeConstraintFunc()`, which internally calls `os::vm_page_size()`.